### PR TITLE
Adding branching details for new flavors

### PR DESCRIPTION
**Pull request by latin-panda**
# Description
This PR adds information about the branch and release process for new cht-android flavors.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: content/en/core/guides/android/branding.md [hunk 1]
```diff
@@ -9,14 +9,14 @@ description: >
 {{% pageinfo %}}
 This tutorial will take you through building a CHT Android Application off the existing wrapper.
 
-The cht-android application is a thin wrapper to load the CHT Core Framework web application in a WebView.
+The CHT-Android application is a thin wrapper to load the CHT Core Framework web application in a WebView.
```
Comment (garethbowen) [inline content/en/core/guides/android/branding.md:12]:
```suggestion
The CHT Android application is a thin wrapper to load the CHT Core Framework web application in a WebView.
```

File: content/en/core/guides/android/releasing.md [hunk 2]
```diff
@@ -37,4 +37,8 @@ All Medic's android projects automatically build, sign, and release builds via G
 
 _Only CHT Android_
 
-If the release only contains a new flavor but is based on the last version, tag it with the same version that the last one plus a number suffix separated with a `-`, e.g. if the last version was `v1.2.3`, tag it as `v1.2.3-1`. In this case there is no need to publish again the reference apps in the Play Store, or to announce the release in the forum if isn't relevant.
+New flavor should have branched off from the [last stable release](https://github.com/medic/cht-android/releases) in CHT-Android repository, for example, if the latest stable release is `v1.2.3` and the branch name is `v1.2.3-new-brand`, then check out the tag and create a branch using the following command:
+ ```
+ git checkout v1.2.3 -b v1.2.3-new-brand
+ ```
+Once the flavor has been added and the pull request has been approved, `squash and merge` the pull request and tag the commit with the same version that the last one plus a number suffix separated with a `-`, e.g. if the last version was `v1.2.3`, tag it as `v1.2.3-1`. In this case there is no need to publish again the reference apps in the Play Store, or to announce the release in the forum if isn't relevant.
```
Comment (latin-panda) [inline content/en/core/guides/android/releasing.md:44]: @garethbowen PR to master/main branch, right?

File: content/en/core/guides/android/branding.md [hunk 3]
```diff
@@ -74,16 +74,22 @@ Once approved you can delete the "test" user, Google conduct the tests only the
 
 Each branded app has an identifier (_id_) that is used to identify and configure it in different parts of the source code and when invoking some commands. In the instructions below we will use as example the id **`new_brand`**.
 
-1. Add `productFlavors { <new_brand> { ... } }` in [build.gradle](https://github.com/medic/cht-android/blob/master/build.gradle), e.g.:
+1. Check out the tag from the [last stable release](https://github.com/medic/cht-android/releases) in CHT-Android repository and create a branch, for example, if the latest stable release is `v0.11.0` and the branch name is `v0.11.0-new-brand`, then the command is:
```
Comment (garethbowen) [inline content/en/core/guides/android/branding.md:77]:
```suggestion
1. Check out the tag from the [last stable release](https://github.com/medic/cht-android/releases) in CHT Android repository and create a branch, for example, if the latest stable release is `v0.11.0` and the branch name is `v0.11.0-new-brand`, then the command is:
```

File: content/en/core/guides/android/branding.md [hunk 4]
```diff
@@ -123,7 +129,7 @@ Each branded app has an identifier (_id_) that is used to identify and configure
            ANDROID_KEY_PASSWORD: ${{ secrets.ANDROID_KEY_PASSWORD_NEW_BRAND }}
    ```
 
-   The variables in the `env` sections point to a keystore and the passwords to unlock the keystore that will be generated in the following steps, but it's important to follow the name convention, in the example all the variables that are configured in Github Actions end with the suffix `_NEW_BRAND`, these variables need to be added in the cht-android repo settings by a manager of Medic.
+   The variables in the `env` sections point to a keystore and the passwords to unlock the keystore that will be generated in the following steps, but it's important to follow the name convention, in the example all the variables that are configured in Github Actions end with the suffix `_NEW_BRAND`, these variables need to be added in the CHT-Android repository settings by a manager of Medic.
```
Comment (garethbowen) [inline content/en/core/guides/android/branding.md:132]:
```suggestion
   The variables in the `env` sections point to a keystore and the passwords to unlock the keystore that will be generated in the following steps, but it's important to follow the name convention, in the example all the variables that are configured in Github Actions end with the suffix `_NEW_BRAND`, these variables need to be added in the CHT Android repository settings by a manager of Medic.
```

File: content/en/core/guides/android/branding.md [hunk 5]
```diff
@@ -219,7 +225,7 @@ Because the files generated here are signed with the same key that you are going
 
 ### 5. Release the new flavor
 
-Once you have your pull request approved in the cht-android repo, it's recommended to create an alpha version before merge it to master and do the final release.
+Once you have your pull request approved in the CHT-Android repository, it's recommended to create an alpha version before merge it to master and do the final release.
```
Comment (garethbowen) [inline content/en/core/guides/android/branding.md:228]:
```suggestion
Once you have your pull request approved in the CHT Android repository, it's recommended to create an alpha version before merge it to master and do the final release.
```

File: content/en/core/guides/android/releasing.md [hunk 6]
```diff
@@ -37,4 +37,8 @@ All Medic's android projects automatically build, sign, and release builds via G
 
 _Only CHT Android_
 
-If the release only contains a new flavor but is based on the last version, tag it with the same version that the last one plus a number suffix separated with a `-`, e.g. if the last version was `v1.2.3`, tag it as `v1.2.3-1`. In this case there is no need to publish again the reference apps in the Play Store, or to announce the release in the forum if isn't relevant.
+New flavor should have branched off from the [last stable release](https://github.com/medic/cht-android/releases) in CHT-Android repository, for example, if the latest stable release is `v1.2.3` and the branch name is `v1.2.3-new-brand`, then check out the tag and create a branch using the following command:
```
Comment (garethbowen) [inline content/en/core/guides/android/releasing.md:40]:
```suggestion
The new flavor should have branched off from the [last stable release](https://github.com/medic/cht-android/releases) in CHT Android repository. For example, if the latest stable release is `v1.2.3` and the branch name is `v1.2.3-new-brand`, then check out the tag and create a branch using the following command:
```

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice additions!

I've made some branding suggestions but otherwise good to go!

File: content/en/core/guides/android/releasing.md (hunk 2 above)
Comment (garethbowen) [reply content/en/core/guides/android/releasing.md:44]: What do you mean?

Comment (latin-panda) [reply content/en/core/guides/android/releasing.md]: We had a chat and the PR made to the release branch, then the commit is backport to master/main

File: content/en/core/guides/android/branding.md (hunk 1 above)
Comment (latin-panda) [reply content/en/core/guides/android/branding.md]: Done

File: content/en/core/guides/android/branding.md (hunk 3 above)
Comment (latin-panda) [reply content/en/core/guides/android/branding.md]: Done

File: content/en/core/guides/android/branding.md (hunk 4 above)
Comment (latin-panda) [reply content/en/core/guides/android/branding.md]: Done

File: content/en/core/guides/android/branding.md (hunk 5 above)
Comment (latin-panda) [reply content/en/core/guides/android/branding.md]: Done

File: content/en/core/guides/android/releasing.md (hunk 6 above)
Comment (latin-panda) [reply content/en/core/guides/android/releasing.md]: Thanks done

**Comment by latin-panda** [comment]
Hi @garethbowen I've added some more details about the pull request. Please have a look, thanks!

File: content/en/core/guides/android/branding.md [hunk 7]
```diff
@@ -219,9 +225,9 @@ Because the files generated here are signed with the same key that you are going
 
 ### 5. Release the new flavor
 
-Once you have your pull request approved in the cht-android repo, it's recommended to create an alpha version before merge it to master and do the final release.
+Make a pull request to the release branch in the CHT Android repository, once approved and before merging, it's recommended to create an alpha version to do final tests before releasing.
```
Comment (garethbowen) [inline content/en/core/guides/android/branding.md:228]: This is now quite long. How about this instead?

1. Make a Pull Request to the release branch in the CHT Android repository.
2. Once approved it's recommended to create an alpha version to do final tests.
3. Merge the PR.
4. [Release the flavor]({{< ref "core/guides/android/releasing#new-flavor-release" >}}).

File: content/en/core/guides/android/branding.md [hunk 8]
```diff
@@ -219,9 +225,9 @@ Because the files generated here are signed with the same key that you are going
...
 
-Checkout the [Release]({{< ref "core/guides/android/releasing#new-flavor-release" >}}) page, where it's explained the different stages and instructions for releasing. The last section _"New flavor release"_ has special instructions of how to release a new brand.
+Checkout the [Release]({{< ref "core/guides/android/releasing#new-flavor-release" >}}) page, where it's explained the different stages and instructions for releasing. The last section _"New flavor release"_ has special instructions about how to release a new brand.
```
Comment (garethbowen) [inline content/en/core/guides/android/branding.md:230]: If you like the suggestion above you can delete this line (it's step 4 above).

```suggestion
```

File: content/en/core/guides/android/releasing.md [hunk 9]
```diff
@@ -37,4 +37,9 @@ All Medic's android projects automatically build, sign, and release builds via G
...
+The new flavor should have branched off from the [last stable release](https://github.com/medic/cht-android/releases) in CHT Android repository, for example, if the latest stable release is `v1.2.3` and the branch name is `v1.2.3-new-brand`, then check out the tag and create a branch using the following command:
+ ```
+ git checkout v1.2.3 -b v1.2.3-new-brand
+ ```
+Add the flavor and make a pull request to the release branch. Once it has been approved, `squash and merge` the pull request, tag the commit with the same version as the last one and add a number suffix separated with a dash character (`-`), e.g. if the last version was `v1.2.3`, tag it as `v1.2.3-1`. In this case, it's not necessary to publish again the reference apps in Play Store or to announce the release in the forum, if not relevant.
+Finally, make sure this commit is backport to the repository's main branch.
```
Comment (garethbowen) [inline content/en/core/guides/android/releasing.md:45]:
```suggestion
Finally, make sure this commit is backported to the repository's main branch.
```

**Review by garethbowen — CHANGES_REQUESTED** [review]
A few suggestions inline to help readability. Nothing major!

File: content/en/core/guides/android/branding.md (hunk 7 above)
Comment (latin-panda) [reply content/en/core/guides/android/branding.md]: Nice done!

**Review by garethbowen — APPROVED** [review]
Awesome, thanks!

**Pull request merged**